data/content_scripts/posting_process/sendkeys.js

```diff
@@ -0,0 +1,766 @@
+// Cross-broswer implementation of text ranges and selections
+// documentation: http://bililite.com/blog/2011/01/17/cross-browser-text-ranges-and-selections/
+// Version: 2.6
+// Copyright (c) 2013 Daniel Wachsstock
+// MIT license:
```

Comment:
Emm.. I want to point out that, the function I posted in the link is only trying to simulate the keyboard event (not text event) in the host page, because Chrome can correctly pass sandboxed text event into the host page but not sandboxed keyboard event.. More accurately, Chrome can pass sandboxed keyboard event but all of the properties (like keyCode) are missing. In embeded-posting, those properties are essential so I have to simulate the keyboard event by inserting inline `<script>`. You may still need to change some code in order to adapt to your situation.

However I think the core idea of the implementation is valueable.

1. It adds a magic property to the target element in the sandboxed environmemt (let's say, content script), which is used to find out the target element in the host environment (inline script). As you can see, by using this magic property, we needn't pass object references between content script and inline script (which is almost not possible as well), instead we can just use plain strings.

2. It executes the code by inserting the content of the function (call `.toString()`), which also minimized extra code. In such case, we needn't hard code (I mean, in `string` form) the function it self or using extra file to store the function content for inline scripts.

3. It uses `JSON.stringify` to manipulate the parameter into the form of inline script, which has better compatibility for different data types. This enables us to pass simple object to the inline script as well. Those literals are well supported by encoding them as JSON.  (notice that the original approach doesn't use this function)